### PR TITLE
Master aktualizr pkgconfig

### PR DESCRIPTION
**Pull request by liuming50**

### CODE REVIEW

**Comment by OYTIS** [comment]
That's quite an improvement, thank you. Let us run our usual tests.

**Comment by OYTIS** [comment]
I'm not sure if installing host-tools, garage-deploy, garage-sign etc. on a board is a valid scenario. Can be quite dangerous (from the backend security point of view) to be honest.

File: recipes-sota/aktualizr/aktualizr_git.bb [hunk 1]
```diff
@@ -32,29 +28,34 @@ BRANCH ?= "master"
...
-                                      -DBUILD_SYSTEMD=OFF \
-                                      -DGARAGE_SIGN_VERSION=${GARAGE_SIGN_VERSION} \
-                                      -DGARAGE_SIGN_SHA256=${GARAGE_SIGN_SHA256}"
+EXTRA_OECMAKE = "-DCMAKE_BUILD_TYPE=Release -DAKTUALIZR_VERSION=${PV}"
+
+GARAGE_SIGN_OPS = "${@ '-DGARAGE_SIGN_VERSION=%s' % d.getVar('GARAGE_SIGN_VERSION') if d.getVar('GARAGE_SIGN_VERSION') is not None else ''} \
+                   ${@ '-DGARAGE_SIGN_SHA256=%s' % d.getVar('GARAGE_SIGN_SHA256') if d.getVar('GARAGE_SIGN_SHA256') is not None else ''} \
+                  "
+
+PACKAGECONFIG ?= "ostree ${@bb.utils.filter('DISTRO_FEATURES', 'systemd', d)} ${@bb.utils.filter('SOTA_CLIENT_FEATURES', 'hsm serialcan ubootenv', d)}"
```
Comment (OYTIS) [inline recipes-sota/aktualizr/aktualizr_git.bb]: We could get rid of `SOTA_CLIENT_FEATURES` alltogether I believe.

File: recipes-sota/aktualizr/aktualizr_git.bb [hunk 2]
```diff
@@ -32,29 +28,34 @@ BRANCH ?= "master"
...
+PACKAGECONFIG_class-native = "sota-tools"
+PACKAGECONFIG[warning-as-error] = "-DWARNING_AS_ERROR=ON,-DWARNING_AS_ERROR=OFF,"
```
[the remark this answers is not in the document]
Comment (liuming50) [reply recipes-sota/aktualizr/aktualizr_git.bb]: Agreed, will send a V3 to remove them.

**Comment by lbonn** [comment]
Thanks! Passed oe-selftests except for a seemingly unrelated raspberrypi error.

**Comment by liuming50** [comment]
Hi, Ibonn:

Thanks for your work, may I know is there any plan to test/merge my same
patches to other branches rocko/sumo/thud, we are currently using rocko and
thud branches and we are being blocked by this issue.

//Ming Liu

lbonn <notifications@github.com> 於 2019年2月7日 週四 上午10:22寫道：

**Comment by patrickvacek** [comment]
> Thanks for your work, may I know is there any plan to test/merge my same patches to other branches rocko/sumo/thud, we are currently using rocko and thud branches and we are being blocked by this issue.

I think we're working on it, but there are a lot of PRs to test and review right now. Yours are in the queue!

**Comment by liuming50** [comment]
Thanks for the update!

//Ming Liu

Patrick Vacek <notifications@github.com> 於 2019年2月11日 週一 下午3:04寫道：

**Pull request merged**